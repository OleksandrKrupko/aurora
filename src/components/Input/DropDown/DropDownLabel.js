import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import { typography } from "../../../theme";
import { TWO_SIZE_VARIANT } from "../../../utils/sizes";

const fontForSizeMap = {
  [TWO_SIZE_VARIANT[0]]: typography.size.hecto,
  [TWO_SIZE_VARIANT[1]]: typography.size.kilo
};

const StyledLabel = styled.label`
  font-size: ${({ size }) => fontForSizeMap[size]};
  opacity: ${({ disabled }) => (disabled ? "0.4" : "1")};
`;

const DropDownLabel = ({ children, size, disabled, ...otherProps }) => (
  <StyledLabel size={size} disabled={disabled} {...otherProps}>
    {children}
  </StyledLabel>
);

DropDownLabel.propTypes = {
  size: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

DropDownLabel.defaultProps = {
  size: TWO_SIZE_VARIANT[1],
  disabled: false
};

export default DropDownLabel;
