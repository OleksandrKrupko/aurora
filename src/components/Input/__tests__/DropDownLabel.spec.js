import React from "react";
import renderer from "react-test-renderer";

import DropDownLabel from "../DropDown/DropDownLabel";

describe("<DropDownLabel />", () => {
  it("renders small size label correctly", () => {
    const component = renderer.create(
      <DropDownLabel size="small">label text</DropDownLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders large size label correctly", () => {
    const component = renderer.create(
      <DropDownLabel>label text</DropDownLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders small disabled label correctly", () => {
    const component = renderer.create(
      <DropDownLabel size="small" disabled>
        label text
      </DropDownLabel>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
