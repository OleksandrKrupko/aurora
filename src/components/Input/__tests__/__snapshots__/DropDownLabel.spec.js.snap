// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownLabel /> renders large size label correctly 1`] = `
<label
  className="muQhD"
  disabled={false}
  size="large"
>
  label text
</label>
`;

exports[`<DropDownLabel /> renders small disabled label correctly 1`] = `
<label
  className="hcMhoQ"
  disabled={true}
  size="small"
>
  label text
</label>
`;

exports[`<DropDownLabel /> renders small size label correctly 1`] = `
<label
  className="dFBmoE"
  disabled={false}
  size="small"
>
  label text
</label>
`;
